c/tree.c

```diff
@@ -1,16 +1,236 @@
 #include "tree.h"
 
+#ifndef WIN32
+//  WIN32 build and link see https://github.com/witwall/mman-win32
 #include <sys/mman.h>
+#endif
+
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
+
+#ifdef WIN32
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#include <_mingw.h>
+#include <windows.h>
+#include <errno.h>
+#include <io.h>
+#endif
+
 #include <fcntl.h>
 #include <errno.h>
 #include <inttypes.h>
 #include <netdb.h>
 #include <stdbool.h>
 #include <stdio.h>
 
+#ifdef WIN32
+	// Source: https://github.com/witwall/mman-win32
```

Comment:
Please split out this code into a separate C file with clear licensing information. It appears that it is under the MIT license.